arima.py

```diff
@@ -0,0 +1,198 @@
+import numpy as np
+import pandas as pd
+from sklearn.metrics import mean_squared_error
+from statsmodels.tsa.stattools import adfuller
+from sklearn.linear_model import LinearRegression
+import statsmodels.api as sm
+
+
+def adf_check(time_series):
+    """
+    Pass in a time series, returns ADF report
+    """
+    result = adfuller(time_series)
+    print('Augmented Dickey-Fuller Test:')
+    labels = ['ADF Test Statistic', 'p-value',
+              'Number of Lags Used', 'Number of Observations Used']
+
+    for value, label in zip(result, labels):
+        print(label+' : '+str(value))
+
+    if result[1] <= 0.05:
+        print("strong evidence against the null hypothesis, reject the null hypothesis. Data has no unit root and is stationary")
+    else:
+        print("weak evidence against null hypothesis, time series has a unit root, indicating it is non-stationary \n")
+
+
+def AR(p, df, key):
+    '''
+    Generating the lagged p terms
+    '''
+    df_temp = df
+
+    for i in range(1, p+1):
+        df_temp['Shifted_values_%d' % i] = df_temp[key].shift(i)
+
+    train_size = (int)(0.8 * df_temp.shape[0])
+
+    # Breaking data set into test and training
+    df_train = pd.DataFrame(df_temp[0:train_size])
+    df_val = pd.DataFrame(df_temp[train_size:df_temp.shape[0]])
+
+    df_train_2 = df_train.dropna()
+    # X contains the lagged values ,hence we skip the first column
+    try:
+        X_train = df_train_2.iloc[:, 1:].values.reshape(-1, p)
+    except:
+        return [pd.DataFrame(), pd.DataFrame(), 0, 0, float("inf")]
+    # Y contains the value,it is the first column
+    y_train = df_train_2.iloc[:, 1].values.reshape(-1, 1)
+
+    # Running linear regression to generate the coefficents of lagged terms
+    model = sm.OLS(y_train, sm.add_constant(X_train)).fit()
+
+    theta = np.array(model.params[1:])
+    intercept = model.params[0]
+    df_train_2['Predicted_Values'] = X_train.dot(theta) + intercept
+    # df_train_2[['Value','Predicted_Values']].plot()
+
+    X_val = df_val.iloc[:, 1:].values.reshape(-1, p)
+    df_val['Predicted_Values'] = X_val.dot(theta) + intercept
+    # df_test[['Value','Predicted_Values']].plot()
+
+    RMSE = np.sqrt(mean_squared_error(
+        df_val[key], df_val['Predicted_Values']))
+    AIC = model.aic
+    BIC = model.bic
+
+    # print("The RMSE is :", RMSE, ", Value of p : ",
+    #       p, ", AIC is : ", AIC, ", BIC is : ", BIC)
+    return [df_train_2, df_val, theta, intercept, RMSE, AIC, BIC]
+
+
+def MA(q, res):
+    ''' Moving Average'''
+
+    for i in range(1, q+1):
```

Comment:
Maybe add a comment for this smaller section of code to explain what's happening